sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -419,106 +551,30 @@ struct tpu_params<tpu::amx, Ta, Tb, Tc, 0, 0, 0,
   template <typename Group, layout LayoutB>
   using joint_matrix_b = joint_matrix<Group, Tb, use::b, K, N, LayoutB>;
   template <typename Group>
-  using joint_matrix_accumulator = joint_matrix<Group, Tc, use::accumulator, M, N>;
-
-  static constexpr uint32_t numtiles = 8;
-  static constexpr scope_t scopes[] = {scope_t::sub_group};
-  static constexpr int num_scopes = sizeof(scopes) / sizeof(scope_t);
-  struct combination {
-    uint32_t max_msize;
-    uint32_t max_nsize;
-    uint32_t max_ksize;
-    uint32_t msize;
-    uint32_t nsize;
-    uint32_t ksize;
-    matrix_type atype;
-    matrix_type btype;
-    matrix_type accumulatortype;
-  };
-  // In this case, the combinations array contain only the combinations that correspond to the Ta, Tb, and Tc 
-  // types that the user provided
-  static constexpr combination combinations[] = {
-      {16, 16, (sizeof(Ta) == 1) ? 64 : 32}};
-  static constexpr int num_combinations =
-      sizeof(combinations) / sizeof(combination);
-};
-
-// General query form:
-// types are not given, no default sizes and no implicit matrix construction
-template <int sM, int sN, int sK>
-struct tpu_params<tpu::amx, void, void, void, sM, sN, sK> {
-  static constexpr uint32_t numtiles = 8;
-  static constexpr scope_t scopes[] = {scope_t::sub_group};
-  static constexpr int num_scopes = sizeof(scopes) / sizeof(scope_t);
-  struct combination {
-    uint32_t max_msize;
-    uint32_t max_nsize;
-    uint32_t max_ksize;
-    uint32_t msize;
-    uint32_t nsize;
-    uint32_t ksize;
-    matrix_type atype;
-    matrix_type btype;
-    matrix_type accumulatortype;
-  };
-  
-  static constexpr combination combinations[] = {
-      {16, 16, 64, 0, 0, 0, matrix_type::sint8, matrix_type::sint8, matrix_type::sint32},
-      {16, 16, 64, 0, 0, 0, matrix_type::sint8, matrix_type::uint8, matrix_type::sint32},
-      {16, 16, 64, 0, 0, 0, matrix_type::uint8, matrix_type::sint8, matrix_type::sint32},
-      {16, 16, 64, 0, 0, 0, matrix_type::uint8, matrix_type::uint8, matrix_type::sint32},
-      {16, 16, 32, 0, 0,0, matrix_type::bf16, matrix_type::bf16, matrix_type::fp32}};
-  static constexpr int num_combinations =
-      sizeof(combinations) / sizeof(combination);
+  using joint_matrix_accumulator = joint_matrix<Group, Tc,
+  use::accumulator, M, N>;
 };
-
-
 enum class tpu {
   xmx8,
   xmx16,
   amx
 };
```

Comment:
Can we remove this and use `sycl::ext::oneapi::experimental::architecture` instead?
See [sycl_ext_oneapi_device_architecture](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_oneapi_device_architecture.asciidoc).

Reply:
[sycl_ext_oneapi_device_architecture](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_oneapi_device_architecture.asciidoc). does not have an enum for the different CPU generations.

Also, the enum we have now for the matrix hardware does not necessarily grow with number of generations of devices unless there is a drastic change in the matrix hardware. For instance, for DG2 and ATS-M, xmx8 will be used for both.

Lastly, the main problem seems to be with creating this dependency on an another experimental extension.  sycl_ext_oneapi_device_architecture states that it only works for AOT compilation and there are limitations with the implementation. It seems to be risky to depend on that.

Reply:
> Lastly, the main problem seems to be with creating this dependency on an another experimental extension. sycl_ext_oneapi_device_architecture states that it only works for AOT compilation and there are limitations with the implementation. It seems to be risky to depend on that.

This should not be a concern.  The enum has no limitations about AOT vs. JIT.  It's just an enum.

> Also, the enum we have now for the matrix hardware does not necessarily grow with number of generations of devices unless there is a drastic change in the matrix hardware. For instance, for DG2 and ATS-M, xmx8 will be used for both.

My concern is that users won't know what generation of the matrix hardware is incorporated into each device.  User's are much more likely to know the name of the device they have.  Rather than forcing our users to know the translation of device names to matrix hardware, I think our extension should do this translation.  Our implementation can have an internal table that does this translation.

We also expect to have a query that allows the user to get the name of a device, returning one of these enums.  We don't have any similar query, which tells the user the matrix hardware generation.

I agree that some extension is needed for the CPU generations.  My plan was to add enumerations to `sycl::ext::oneapi::experimental::architecture` that follow the compilation options supported by the `opencl-aot` tool.  Last I looked these were "avx512", "avx2", "avx", and "sse4.2".  Are we adding some similar options for the matrix hardware?

Reply:
The compilation options when using AMX intrinsics are: -mamx-tile -mamx-int8 -mamx-bf16

Reply:
@gmlueck, we use -march=sapphirerapids for AOT compilation. So CPU enumeration would follow the same logic as the GPU one.